specifyweb/frontend/js_src/lib/attachmentplugin.js

```diff
@@ -34,15 +35,16 @@ module.exports =  UIPlugin.extend({
             return this;
         },
         addAttachment: function() {
-            this.$el.append('<form enctype="multipart/form-data"><input type="file" name="file"></form>');
-            this.$('input').click();
+            this.$el.append('<form enctype="multipart/form-data"><input type="checkbox" id="isPublicImage">  Is this a public image file?<br><input type="file" name="file"></form>');
```

Comment:
The checkbox should probably only be included if the public image server is configured.

Also, it seems like the UI should force the user to pick public/private before choosing the file, because the file gets uploaded immediately (which is admittedly not ideal either, on my part).